nats-base-client/servers.ts

```diff
@@ -36,12 +37,20 @@ function hostPort(
   // that means that protocols other than HTTP/S are not
   // parsable correctly.
   const url = new URL(`http://${u}`);
+
+  let sport = "";
   if (!url.port) {
-    url.port = `${DEFAULT_PORT}`;
+    if (u.endsWith(":80")) {
```

Comment:
Is there not a `URL` function that parses host and port for you? It looks strange here that you look for :80 or :443. I would simply try to parse the port, and if there is no port, use the DEFAULT_PORT.
Note that what we have done for websocket in the Go library is to say that if protocol is "ws" then default port is 80, and 443 for wws, any other is 4222.

Reply:
Yes no such thing - the websocket default ports etc are handled in the websocket library. There will be some work there as well.